### DISCUSSION TITLE
Multiple app using Elsa with the same storage

### DISCUSSION THREAD

**Discussion by Alexplose** [elsa-workflows/elsa-core#1530] [Q&A] [answered]
I'm using Elsa in a monolithic web app, so it hosts Elsa, the Dashboard, my app and some backoffice stuff.

I'm planning to separate my app in two apps.

The first one will host Elsa and my app.
The second one will host Elsa + Dashboard and backoffice functions.

The two apps will interact with Elsa, in the same storage.

Will it work or it's better to host Elsa+Dashboard in a dedicated app, and make my other client applications to interact with it ?

**Comment by sfmskywalker** [comment, marked as answer]
That should work just fine. 

But assuming that your Backoffice app doesn't use workflows, it might be better to keep the Dashboard+Backoffice separate from the Workflow+MyApp Server, allowing you to deploy & scale out the Dashboard+Backoffice application separately from the Workflow+MyApp application.

Additionally, you could keep the Dashboard entirely separate from your Backoffice as well so that they are separately deployable & scalable too.

**Reply by Alexplose** [reply]
Ok so no problem triggering workflows from one side or the other,  what I mean by triggering is calling : 
await _signaler.TriggerSignalAsync("Signal", correlationId: CorrelationId);
from My App or BackOffice since they hosts Elsa, and target same database.
Or I should change all my calls to Elsa api calls ?

**Reply by sfmskywalker** [reply]
No problem, but since you will have (at least) two Elsa Workflow hosts (one residing in My App and another in BackOffice) you will probably want to configure the two for a multi-node setup (since you are effectively running at least two nodes now). Details on how to do that can be found [here](https://elsa-workflows.github.io/elsa-core/docs/next/hosting/hosting-distributed-hosting).

**Reply by Alexplose** [reply]
Thanks for your reply

**Reply by Alexplose** [reply]
As a follow up, any tips to work locally (dev environment) without the need of installing rabbitMQ or create Azure service bus.

**Reply by sfmskywalker** [reply]
Sure. For local development, you don't need either. Rebus is setup with an in-memory provider by default. However, if you want to test multi-node execution on your local machine, you will nee to use  shared resource such as RabbitMQ, Redis or otherwise.

But that's easy too when using Docker and docker compose.

Look at the root folder where you will find a docker-compose.yaml file. This file defines a number of services that you can just start individually, such as RabbitMQ and Redis.

You can do so from the CLI with `docker-compose`, but since I'm spoiled with using Rider IDE, I simply click the play buttons to start them:

![image](https://user-images.githubusercontent.com/938393/134035743-1832bdc6-56d1-4dc8-a83c-d7bd8bd4238a.png)

**Reply by Alexplose** [reply]
That's working fine, thanks for the detailled reply.
